src/main/java/com/commercetools/emailprocessor/email/EmailProcessor.java

```diff
@@ -0,0 +1,91 @@
+package com.commercetools.emailprocessor.email;
+
+import com.commercetools.emailprocessor.model.Email;
+import com.commercetools.emailprocessor.model.Statistics;
+import com.commercetools.emailprocessor.model.TenantConfiguration;
+import io.sphere.sdk.client.SphereClient;
+import io.sphere.sdk.customobjects.CustomObject;
+import io.sphere.sdk.customobjects.queries.CustomObjectQuery;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.HttpStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+
+public class EmailProcessor {
+
+
+    private static final Logger LOG = LoggerFactory.getLogger(EmailProcessor.class);
+    private static final String PARAM_EMAIL_ID = "emailid";
+    private static final String PARAM_TENANT_ID = "tenantid";
+    private static final String CONTAINER_ID = "unprocessedEmail";
+  ;
+    private static final String EMAIL_STATUS_PENDING = "pending";
+
+    public EmailProcessor() {
+    }
+
+    /**
+     * Iterate through all Email objects and triggers the webhook for each  pending email object
+     *
+     * @param tenantConfiguration Configuration of a tenant
+     * @return Statics of the sended emails
+     */
+
+    public Statistics processEmails(TenantConfiguration tenantConfiguration) {
+
+        SphereClient client = tenantConfiguration.getSphereClient();
+        CustomObjectQuery<Email> query = CustomObjectQuery.of(Email.class);
+        query = query.byContainer(CONTAINER_ID);
+        return client.execute(query).thenApply(response -> {
```

Comment:
This query by default returns only 20 first elements, right? Don't you need to iterate through all pages?

Reply:
What if you have more than 100 items?

Reply:
The rest of the email objects will be processed in the next job run? or should I implement a pagination?

Reply:
https://github.com/commercetools/commercetools-email-retry-processor/issues/6